packages/next/server/lib/start-server.ts

```diff
@@ -18,8 +19,21 @@ export default async function start(
   requestHandler = app.getRequestHandler()
 
   await new Promise<void>((resolve, reject) => {
-    // This code catches EADDRINUSE error if the port is already in use
-    srv.on('error', reject)
+    const isExplicitPort = serverOptions.inlinePort != null
+    srv.on('error', (err: NodeJS.ErrnoException) => {
+      // This code catches EADDRINUSE error if the port is already in use
+      if (
+        err.code === 'EADDRINUSE' &&
+        serverOptions.isNextDevCommand &&
+        !isExplicitPort
+      ) {
+        warn(`Port ${port} is in use, trying ${port! + 1} instead.`)
+        port! += 1
+        srv.listen(port, hostname)
```

Comment:
This should probably only retry 10 times or so before failing, otherwise it could be stuck in an infinite loop